pkg/testframework/clientset.go

```diff
@@ -0,0 +1,62 @@
+package testframework
+
+import (
+	"fmt"
+	"testing"
+
+	clientappsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	clientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+
+	clientconfigv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/console-operator/pkg/client"
+	clientconsolev1alpha1 "github.com/openshift/console-operator/pkg/generated/clientset/versioned/typed/console/v1alpha1"
+)
+
+// Clientset is a set of Kubernetes clients.
+type Clientset struct {
+	clientcorev1.CoreV1Interface
+	clientappsv1.AppsV1Interface
+	clientconfigv1.ConfigV1Interface
+	clientconsolev1alpha1.ConsoleV1alpha1Interface
+}
+
+// NewClientset creates a set of Kubernetes clients. The default kubeconfig is
+// used if not provided.
+func NewClientset(kubeconfig *restclient.Config) (clientset *Clientset, err error) {
+	if kubeconfig == nil {
+		kubeconfig, err = client.GetConfig()
+		if err != nil {
+			return nil, fmt.Errorf("unable to get kubeconfig: %s", err)
+		}
+	}
+
+	clientset = &Clientset{}
+	clientset.CoreV1Interface, err = clientcorev1.NewForConfig(kubeconfig)
```

Comment:
We want to swallow all of the`err` below?  Should we return the `nil, err` instead?

Reply:
the `err` is not shallowed. Checked the function return values.
```go
(clientset *Clientset, err error)
```

they are going to be returned.

Reply:
Right. I think we prob want to avoid bare returns.  So far I have in this repo, generally seems the consensus is against bare returns (though they still exist in some older code in origin).

Reply:
I should not have said "swallowed", my bad.

Reply:
no worries ! ... if we dont use the default returns across the codebase lets just dont use them ! :+1: